src/PipelineNet/ChainsOfResponsibility/AsyncResponsibilityChain.cs

```diff
@@ -142,20 +153,33 @@ public async Task<TReturn> Execute(TParameter parameter, CancellationToken cance
                         }
                     }
 
-                    if (resolverResult.IsDisposable && !(middleware is IDisposable
+                    if (resolverResult == null || resolverResult.Middleware == null)
+                    {
+                        throw new InvalidOperationException($"'{MiddlewareResolver.GetType()}' failed to resolve middleware of type '{type}'.");
+                    }
+
+                    if (resolverResult.IsDisposable && !(resolverResult.Middleware is IDisposable
 #if NETSTANDARD2_1_OR_GREATER
```

Comment:
I'll create a PR to drop support for net standard < 2.1 for the version 1.0.0 of the package. This will help to simplify the code.

Reply:
Don't do that please. 2.1 is kind of uncommon. And it doesn't work on `netframework` at all.

Reply:
Oh, I thought the 2.1 was the one that was also used by .net framework, just saw it was the 2.0.
It's a shame that the full framework still has no end of life support.

Ok, we can at least bump it to [2.0](https://learn.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-0#select-net-standard-version), although that wouldn't change anything in terms of code.

Reply:
Why would you do that then? It would only break libraries that target 1.1-1.6.

`PipelineNet.ServiceProvider` needs to be 2.0 thoguh because it has a dependency on `Microsoft.Extensions.DependencyInjection.Abstractions.2.1.0`.

Reply:
The [recommendation](https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/9.0/netstandard-warning) from Microsoft is to have the netstandard 2.0 as the lowest version for packages unless there's a good use case to have for supporting old ones.

Do you need it for older Net Framework versions?